In [11]:

import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

data = {

  'luas_m2': [21, 36, 45, 50, 60, 72, 85, 90, 100, 120],          # Blok 1 (X1)
    'jarak_pusat_kota': [15, 12, 10, 8, 7, 5, 4, 3, 2, 1],         # Blok 2 (X2)
    'fasilitas_skor': [5, 6, 6, 7, 7, 8, 8, 9, 9, 10],            # Blok 3 (X3)
    'keputusan_sewa': [0, 0, 0, 1, 0, 1, 1, 1, 1, 1]
}

df = pd.DataFrame(data)
print(df)

   luas_m2  jarak_pusat_kota  fasilitas_skor  keputusan_sewa
0       21                15               5               0
1       36                12               6               0
2       45                10               6               0
3       50                 8               7               1
4       60                 7               7               0
5       72                 5               8               1
6       85                 4               8               1
7       90                 3               9               1
8      100                 2               9               1
9      120                 1              10               1


Pada praktikum ini, saya membuat data mandiri untuk melakukan klasifikasi terhadap keputusan penyewa menggunakan total empat blok data: tiga blok fitur sebagai input (luas bangunan, jarak ke pusat kota, dan skor fasilitas) serta satu blok sebagai target (keputusan sewa). Data terdiri dari 10 sampel dengan label biner: 1 untuk "Jadi Sewa" dan 0 untuk "Batal Sewa". Tema ini dipilih karena sangat relevan untuk memodelkan pilihan konsumen. Regresi Logistik digunakan karena output yang ingin dihasilkan adalah klasifikasi biner, bukan nilai angka kontinu.

In [12]:

X = df[['luas_m2', 'jarak_pusat_kota', 'fasilitas_skor']]
y = df['keputusan_sewa']

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

Data dibagi menjadi 80% data training (12 data) dan 20% data testing (3 data) menggunakan train_test_split. Model dilatih menggunakan LogisticRegression() dari library sklearn dengan tiga variabel independen:


* luas_m2 : luas apartemen dalam meter
* jarak_pusat_kota : jaraknya (15-1)
* fasilitas_skor : dari angka 5-10





In [13]:

print("Intercept:", model.intercept_)
print("Koefisien:", model.coef_)

Intercept: [-1.67959909]
Koefisien: [[ 0.03787716 -0.39794771  0.3755044 ]]


Model menghasilkan satu intercept dan tiga koefisien untuk masing-masing variabel independen (luas_m2, jarak_pusat_kota, dan fasilitas_skor).
Intercept: Bernilai -1.67959909, yang merupakan nilai dasar log-odds keputusan sewa ketika semua variabel fitur bernilai nol.

Koefisien positif ditemukan pada variabel luas_m2 (0.03787716) dan fasilitas_skor (0.3755044).
Koefisien negatif ditemukan pada variabel jarak_pusat_kota (-0.39794771).

In [14]:

y_pred = model.predict(X_test)

print("Prediksi:", y_pred)
print("Aktual:", y_test.values)

Prediksi: [1 0]
Aktual: [1 0]


Model memprediksi label (0 atau 1) untuk 3 data testing dan dibandingkan dengan nilai aktualnya. Jika prediksi dan aktual sama, berarti model berhasil mengklasifikasikan dengan benar.